src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_SIZE = 16;
+    public static final double RESIZE_FACTOR = 0.75;
+    public static final double EXPANSION_COEFFICIENT = 2;
+    private Node<K, V>[] storage;
+    private int size;
+
+    public MyHashMap() {
+        storage = (Node<K, V>[]) new Node[DEFAULT_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
+        checkSize();
+        if (key == null) {
+            if (storage[0] == null) {
+                storage[0] = new Node<>(null, value);
+                size++;
+                return;
+            }
+            if (storage[0].key == null) {
+                storage[0].value = value;
+                return;
+            }
+            System.arraycopy(storage, 0, storage, 1, storage.length - 1);
```

Comment:
in this task we dont need to use `System.arraycopy`. please watch again video about [HashMap structure](https://mate.academy/learn/java-core/java-hash-map?section=video&testTaskSlug=java_hash_map&videoId=619)  and refactor your code